### PR TITLE
Remove duplicated tests from the runtests script.

### PR DESCRIPTION
**Pull request by antiagainst**
Offspring of https://github.com/KhronosGroup/glslang/pull/261.

Quoting @johnkslang, before merging this PR, the following statements need to be true:

- [x] we see Travis successfully executing gtests
- [x] we see general use of gtests with submitters
- [x] we have working Windows instructions for gtests plus runtests

### CODE REVIEW

**Comment by johnkslang** [comment]
Some side questions in the meantime about using gtest:

- In Visual Studio, glslang proper is now totally overwhelmed by "gtest...", "gmock...", and "sample..." projects that camouflage the actual compiler.  gtest is making it harder to do actual development this way.  Can that be pushed one level down or otherwise hidden away?

-  An advantage of the way runtest was set up to use the `install`, whether debug/release/etc.  It automatically tested the most recent binary.  gtest either requires manually specifying everytime, or writing a script that will accidentally test the wrong one.  That is, gtest is more error prone.  Can it similarly use the installed binary instead?

- Is gtest spending most of its time testing something other than glslang?  Sorry I did not yet follow what the first 50+ tests are for.  But, they are taking real time, which I thought one reason to go to gtest was it would not take that much time.

Thanks.

**Comment by antiagainst** [comment]
> * In Visual Studio, glslang proper is now totally overwhelmed by "gtest...", "gmock...", and "sample..." projects that camouflage the actual compiler. gtest is making it harder to do actual development this way. Can that be pushed one level down or otherwise hidden away?

I don't have much experience with VS; and huge thanks to @AWoloszyn, he has sent out PR #268 that groups build targets and hides gtest related stuff from cmake-gui.

> * An advantage of the way runtest was set up to use the install, whether debug/release/etc. It automatically tested the most recent binary. gtest either requires manually specifying everytime, or writing a script that will accidentally test the wrong one. That is, gtest is more error prone. Can it similarly use the installed binary instead?

Also in #268, `glslangtests` is installed into `install/bin`, alongside with `glslangValidator`. By **directly** running this binary, all the gtest-backed tests will be checked. Hope this addresses your concern.

Gtest is unit testing framework; it does its work by calling APIs in glslang and comparing the results. It does not invoke compiled binaries like the way `runtests` does, so the build type of the compiled binaries doesn't mattter at all. Gtest itself doesn't need manually specifying about where to find shader files and expected results; they are given by CMake. And gtest doesn't write any script during testing.

The exact gtest flow is like:

When building, CMake tells gtest the absolute path to `Test` and `Test/baseResults`, and gtest hard-codes these paths in the `glslangtests` binary. When the `glslangtests` binary is invoked (either directly or via `ctest`), it finds the files registered in the lists of various `*.FromFile.cpp` files. For each file, it calls the APIs (`Tshader.parse()`, `glslangToSpv()`) to get it compiled. Then it compares the actual result against the golden files under `Test/baseResults`. Error out if different. So no binaries (except `glslangtests` itself) involved at all.

> * Is gtest spending most of its time testing something other than glslang? Sorry I did not yet follow what the first 50+ tests are for. But, they are taking real time, which I thought one reason to go to gtest was it would not take that much time.

I'm not quite sure I get your concerns here. All the tests run by gtest are legit tests previously run by `runtests` script. Maybe things are different on Windows? Would you mind to paste the log?

**Comment by johnkslang** [comment]
Thanks for the answers.

Here is a log:

```
$ ctest -C Release
      Start  1: gmock-actions_test
 1/57 Test  #1: gmock-actions_test ......................   Passed    0.05 sec
      Start  2: gmock-cardinalities_test
 2/57 Test  #2: gmock-cardinalities_test ................   Passed    0.02 sec
      Start  3: gmock_ex_test
 3/57 Test  #3: gmock_ex_test ...........................   Passed    0.02 sec
      Start  4: gmock-generated-actions_test
 4/57 Test  #4: gmock-generated-actions_test ............   Passed    0.02 sec
      Start  5: gmock-generated-function-mockers_test
 5/57 Test  #5: gmock-generated-function-mockers_test ...   Passed    0.01 sec
      Start  6: gmock-generated-internal-utils_test
 6/57 Test  #6: gmock-generated-internal-utils_test .....   Passed    0.02 sec
      Start  7: gmock-generated-matchers_test
 7/57 Test  #7: gmock-generated-matchers_test ...........   Passed    0.02 sec
      Start  8: gmock-internal-utils_test
 8/57 Test  #8: gmock-internal-utils_test ...............   Passed    0.04 sec
      Start  9: gmock-matchers_test
 9/57 Test  #9: gmock-matchers_test .....................   Passed    0.70 sec
      Start 10: gmock-more-actions_test
10/57 Test #10: gmock-more-actions_test .................   Passed    0.01 sec
      Start 11: gmock-nice-strict_test
11/57 Test #11: gmock-nice-strict_test ..................   Passed    0.03 sec
      Start 12: gmock-port_test
12/57 Test #12: gmock-port_test .........................   Passed    0.02 sec
      Start 13: gmock-spec-builders_test
13/57 Test #13: gmock-spec-builders_test ................   Passed    0.03 sec
      Start 14: gmock_link_test
14/57 Test #14: gmock_link_test .........................   Passed    0.02 sec
      Start 15: gmock_test
15/57 Test #15: gmock_test ..............................   Passed    0.01 sec
      Start 16: gmock-more-actions_no_exception_test
16/57 Test #16: gmock-more-actions_no_exception_test ....   Passed    0.02 sec
      Start 17: gmock_no_rtti_test
17/57 Test #17: gmock_no_rtti_test ......................   Passed    0.06 sec
      Start 18: gtest-death-test_test
18/57 Test #18: gtest-death-test_test ...................   Passed    0.31 sec
      Start 19: gtest_environment_test
19/57 Test #19: gtest_environment_test ..................   Passed    0.02 sec
      Start 20: gtest-filepath_test
20/57 Test #20: gtest-filepath_test .....................   Passed    0.02 sec
      Start 21: gtest-linked_ptr_test
21/57 Test #21: gtest-linked_ptr_test ...................   Passed    0.02 sec
      Start 22: gtest-listener_test
22/57 Test #22: gtest-listener_test .....................   Passed    0.01 sec
      Start 23: gtest_main_unittest
23/57 Test #23: gtest_main_unittest .....................   Passed    0.02 sec
      Start 24: gtest-message_test
24/57 Test #24: gtest-message_test ......................   Passed    0.02 sec
      Start 25: gtest_no_test_unittest
25/57 Test #25: gtest_no_test_unittest ..................   Passed    0.02 sec
      Start 26: gtest-options_test
26/57 Test #26: gtest-options_test ......................   Passed    0.02 sec
      Start 27: gtest-param-test_test
27/57 Test #27: gtest-param-test_test ...................   Passed    0.03 sec
      Start 28: gtest-port_test
28/57 Test #28: gtest-port_test .........................   Passed    3.11 sec
      Start 29: gtest_pred_impl_unittest
29/57 Test #29: gtest_pred_impl_unittest ................   Passed    0.05 sec
      Start 30: gtest_premature_exit_test
30/57 Test #30: gtest_premature_exit_test ...............   Passed    0.04 sec
      Start 31: gtest-printers_test
31/57 Test #31: gtest-printers_test .....................   Passed    0.04 sec
      Start 32: gtest_prod_test
32/57 Test #32: gtest_prod_test .........................   Passed    0.04 sec
      Start 33: gtest_repeat_test
33/57 Test #33: gtest_repeat_test .......................   Passed    0.11 sec
      Start 34: gtest_sole_header_test
34/57 Test #34: gtest_sole_header_test ..................   Passed    0.01 sec
      Start 35: gtest_stress_test
35/57 Test #35: gtest_stress_test .......................   Passed    0.17 sec
      Start 36: gtest-test-part_test
36/57 Test #36: gtest-test-part_test ....................   Passed    0.03 sec
      Start 37: gtest_throw_on_failure_ex_test
37/57 Test #37: gtest_throw_on_failure_ex_test ..........   Passed    0.01 sec
      Start 38: gtest-typed-test_test
38/57 Test #38: gtest-typed-test_test ...................   Passed    0.04 sec
      Start 39: gtest_unittest
39/57 Test #39: gtest_unittest ..........................   Passed    0.15 sec
      Start 40: gtest-unittest-api_test
40/57 Test #40: gtest-unittest-api_test .................   Passed    0.02 sec
      Start 41: gtest-death-test_ex_nocatch_test
41/57 Test #41: gtest-death-test_ex_nocatch_test ........   Passed    0.03 sec
      Start 42: gtest-death-test_ex_catch_test
42/57 Test #42: gtest-death-test_ex_catch_test ..........   Passed    0.04 sec
      Start 43: gtest_no_rtti_unittest
43/57 Test #43: gtest_no_rtti_unittest ..................   Passed    0.14 sec
      Start 44: gtest_break_on_failure_unittest
44/57 Test #44: gtest_break_on_failure_unittest .........   Passed    0.10 sec
      Start 45: gtest_catch_exceptions_test
45/57 Test #45: gtest_catch_exceptions_test .............   Passed    0.06 sec
      Start 46: gtest_color_test
46/57 Test #46: gtest_color_test ........................   Passed    0.14 sec
      Start 47: gtest_env_var_test
47/57 Test #47: gtest_env_var_test ......................   Passed    0.12 sec
      Start 48: gtest_filter_unittest
48/57 Test #48: gtest_filter_unittest ...................   Passed    1.05 sec
      Start 49: gtest_help_test
49/57 Test #49: gtest_help_test .........................   Passed    0.09 sec
      Start 50: gtest_list_tests_unittest
50/57 Test #50: gtest_list_tests_unittest ...............   Passed    0.07 sec
      Start 51: gtest_output_test
51/57 Test #51: gtest_output_test .......................   Passed    0.16 sec
      Start 52: gtest_shuffle_test
52/57 Test #52: gtest_shuffle_test ......................   Passed    0.11 sec
      Start 53: gtest_throw_on_failure_test
53/57 Test #53: gtest_throw_on_failure_test .............   Passed    0.08 sec
      Start 54: gtest_uninitialized_test
54/57 Test #54: gtest_uninitialized_test ................   Passed    0.06 sec
      Start 55: gtest_xml_outfiles_test
55/57 Test #55: gtest_xml_outfiles_test .................   Passed    0.06 sec
      Start 56: gtest_xml_output_unittest
56/57 Test #56: gtest_xml_output_unittest ...............   Passed    0.12 sec
      Start 57: glslang-gtests
57/57 Test #57: glslang-gtests ..........................   Passed    0.61 sec
```

**Comment by antiagainst** [comment]
@johnkslang: Ah, these are tests for the gtest framework itself. Please make sure to avoid checking anything related to gtest when configuring using cmake-gui. (With #268, they are not visible anymore.)

Or, instead of using `ctest`, just run the compiled `glslangtests` binary. That's much easier and much clearer.

**Comment by johnkslang** [comment]
Got it, thanks.

File: Test/runtests [hunk 1]
```diff
@@ -10,8 +10,6 @@ mkdir -p localResults
 # configuration file tests
 #
 echo running configuration file test
-$EXE -c > $TARGETDIR/test.conf
-diff -b $BASEDIR/test.conf $TARGETDIR/test.conf || HASERROR=1
```
Comment (antiagainst) [inline Test/runtests]: This check is covered via `gtests/BuiltInResource.FromFile.cpp`.

**Comment by johnkslang** [comment]
I think we are seeing uses of gtest, and we've waiting long enough, that we can remove the duplication of tests between gtest and runtests.

**Comment by antiagainst** [comment]
Sure! I'll rebase this PR.

**Comment by antiagainst** [comment]
Rebased and checked that # tests in the original txt lists are <= # tests in the gtest lists.

**Comment by johnkslang** [comment]
One question: Does this remove almost all testing done by one of the current bots?  If so, it might be nice (even in general) to keep one test from each category, just as a sanity check.

**Comment by antiagainst** [comment]
@johnkslang:

This PR just stops registering many existing tests to the `runtests` script, but they are still run by gtests. Appveyor & Travis will still exercise the existing tests via the gtests path.

Sorry if the above doesn't answer your question. I think I don't quite get your thinking here.

**Comment by johnkslang** [comment]
> Appveyor & Travis will still exercise the existing tests via the gtests path.

That answers my question, thanks.

**Comment by antiagainst** [comment]
Cool, thanks! :D

**Pull request merged**